### PR TITLE
Port more CLI tests to Python 3

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3691

### CODE REVIEW

**Review by exarkun — DISMISSED** [review]
Thanks.

I gather `test_cli.py` is still not fully ported, thus no addition to the ported list?

Also the news fragment is missing.  Apart from that, looks good, please merge.

**Comment by exarkun** [comment]
> Thanks.
> 
> I gather `test_cli.py` is still not fully ported, thus no addition to the ported list?
> 
> Also the news fragment is missing. Apart from that, looks good, please merge.

Uh.  Scratch all of that, please.  GitHub only showed me a tiny portion of the changes until after I submitted this comment.  Review not yet complete...

File: src/allmydata/scripts/tahoe_ls.py [hunk 1]
```diff
@@ -64,13 +64,14 @@ def list(options):
         print(quote_output(data, quotemarks=False), file=stderr)
         return 1
 
+    path = unicode(path, "utf-8")
     nodetype, d = parsed
     children = {}
     if nodetype == "dirnode":
         children = d['children']
     else:
         # paths returned from get_alias are always valid UTF-8
```
Comment (exarkun) [inline src/allmydata/scripts/tahoe_ls.py:73]: This comment should be moved up to line 67ish

File: src/allmydata/scripts/tahoe_ls.py [hunk 2]
```diff
@@ -64,13 +64,14 @@ def list(options):
         print(quote_output(data, quotemarks=False), file=stderr)
         return 1
 
+    path = unicode(path, "utf-8")
```
Comment (exarkun) [inline src/allmydata/scripts/tahoe_ls.py:67]: There is one use of `path` earlier in this function.  Is it okay or are is it just not exercised by any of the tests (or at least the tests being ported here)?

File: src/allmydata/scripts/tahoe_mv.py [hunk 3]
```diff
@@ -47,6 +49,7 @@ def mv(options, mode="move"):
     if path:
         to_url += "/" + escape_path(path)
 
+    from_path = unicode(from_path, "utf-8")
```
Comment (exarkun) [inline src/allmydata/scripts/tahoe_mv.py:52]: Similar pattern here as in `tahoe_ls.py` where this value is used once above before it is converted.  All good or just untested?

**Review by exarkun — APPROVED** [review]
Thanks.  Generally looks good.  A few comments inline - a couple about possibly missing test coverage.  If those lead to a bunch more work, feel free to re-submit for another review.  Or just split that work off into another ticket if you want.  If they're minor, feel free to merge.

**Comment by itamarst** [comment]
`test_cli.py` is big enough that it's going to be in next PR.

File: src/allmydata/scripts/tahoe_ls.py (hunk 1 above)
Comment (itamarst) [reply src/allmydata/scripts/tahoe_ls.py:73]: Turns out `escape_path()` can take both unicode and bytes, and that branch is actually covered. But yeah should be moved.

**Comment by itamarst** [comment]
Oh, and that change to `test_cli.py` (which is a fine change) should've been in the follow-up branch. That just makes a test or two pass and there are 60 more to get through.

**Pull request merged**